src/routes/transactions/transactions-history.imitation-transactions.controller.spec.ts

```diff
@@ -100,1393 +105,2586 @@ describe('Transactions History Controller (Unit) - Imitation Transactions', () =
-                  value: multisigTransfer.value,
+                type: 'TRANSACTION',
+              },
+              {
+                conflictType: 'None',
+                transaction: {
+                  executionInfo: {
+                    confirmationsRequired: 1,
+                    confirmationsSubmitted: 1,
+                    missingSigners: null,
+                    nonce: multisigTransaction.nonce,
+                    type: 'MULTISIG',
+                  },
+                  id: `multisig_${safe.address}_${multisigTransaction.safeTxHash}`,
+                  safeAppInfo: null,
+                  timestamp: 1710927685000,
+                  txInfo: {
+                    direction: 'OUTGOING',
+                    humanDescription: null,
+                    recipient: {
+                      logoUri: null,
+                      name: null,
+                      value: multisigTransfer.to,
+                    },
+                    richDecodedInfo: null,
+                    sender: {
+                      logoUri: null,
+                      name: null,
+                      value: safe.address,
+                    },
+                    transferInfo: {
+                      decimals: multisigToken.decimals,
+                      imitation: false,
+                      logoUri: multisigToken.logoUri,
+                      tokenAddress: multisigToken.address,
+                      tokenName: multisigToken.name,
+                      tokenSymbol: multisigToken.symbol,
+                      trusted: null,
+                      type: 'ERC20',
+                      value: multisigTransfer.value,
+                    },
+                    type: 'Transfer',
+                  },
+                  txStatus: 'SUCCESS',
+                  txHash: multisigTransaction.transactionHash,
                 },
-                type: 'Transfer',
+                type: 'TRANSACTION',
               },
-              txStatus: 'SUCCESS',
-              txHash:
-                imitationOutgoingTransaction.transfers![0].transactionHash,
-            },
-            type: 'TRANSACTION',
-          },
-          {
-            conflictType: 'None',
-            transaction: {
-              executionInfo: {
-                confirmationsRequired: 1,
-                confirmationsSubmitted: 1,
-                missingSigners: null,
-                nonce: notImitatedMultisigTransaction.nonce,
-                type: 'MULTISIG',
+            ]);
+          });
+      });
+
+      it('should not filter out imitation incoming/outgoing transfers of vanity with an intolerant value within the lookup distance', async () => {
+        const results = [
+          valueIntolerantIncomingTransaction,
+          multisigTransaction,
+          valueIntolerantOutgoingTransaction,
+          notImitatedMultisigTransaction,
+          multisigTransaction,
+        ];
+        networkService.get.mockImplementation(({ url }) => {
```

Comment:
Nit: I have the same concern here. Could we move these input values elsewhere to keep the focus of the test on the scenarios we’re actually testing?

Reply:
See https://github.com/safe-global/safe-client-gateway/pull/2024#discussion_r1810773000.